File: look-and-say/look-and-say-java/src/main/java/pl/frognar/Sequence.java
package pl.frognar;

public class Sequence {
    public static String lookAndSay(String input) {
        var result = "";
        var count = 1;
        for (var index = 0; index < input.length() - 1; index++) {
            var currentDigit = input.charAt(index);
            var nextDigit = input.charAt(index + 1);
            if (currentDigit == nextDigit) {
                count++;
            }
            else {
                result += "%s%s".formatted(count, currentDigit);
                count = 1;
            }
        }

        result += "%s%s".formatted(count, input.charAt(input.length() - 1));
        return result;
    }
}

File: look-and-say/look-and-say-py/03/sequence.py
def look_and_say(input_value: str) -> str:
    result = ''
    count = 1
    index = 0
    if len(input_value) == 2:
        current_digit = input_value[index]
        if current_digit == input_value[index + 1]:
            count += 1
        else:
            result += f'{count}{current_digit}'
    result += f'{count}{input_value[-1]}'
    return result
